### ISSUE TITLE
[A11y] Double Tap brings the `Slider` thumb to the center of the widget. 

### ISSUE THREAD

**Issue by dsanagustin** [flutter/flutter#156427]
### Steps to reproduce

1. Turn on Android TalkBack or iOS VoiceOver.
2. Load the simple app with one Flutter Slider.
3. Try double tapping on a tick.

### Expected results

Double tapping a tick should move the slider indicator to that tick.

### Actual results

If the indicator is off-center, the indicator will return to center after a double tap.

From my own research, I believe this could be related to how `GestureDetectors` and `GestureRecognizers` work when accessibility mode is on.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  MyApp();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Slider A11Y Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text(title)),
      body: MySlider(),
    );
  }
}

class MySlider extends StatefulWidget {
  MySlider({Key? key}) : super(key: key);

  @override
  State createState() => _MySliderState();
}

class _MySliderState extends State<MySlider> {
  double _currentSliderValue = 50.0;

  void _onChanged(double value) {
    setState(() {
      _currentSliderValue = value;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Slider(
      value: _currentSliderValue,
      min: -100,
      max: 100,
      divisions: 8,
      label: 'Current Value: $_currentSliderValue',
      onChanged: _onChanged,
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://youtube.com/shorts/VovzhuLa27s

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
$ fg3 doctor -v
[   +2 ms] ProcessManager.runSync('which java')
[   +5 ms] executing: /usr/local/buildtools/java/jdk/bin/java --version
[   +1 ms] ProcessManager.runSync('/usr/local/buildtools/java/jdk/bin/java --version')
[  +38 ms] Exit code 0 from: /usr/local/buildtools/java/jdk/bin/java --version
[   +1 ms] openjdk 23 2024-10-02
           OpenJDK Runtime Environment (build 23+-google-release-sts-681679659)
           OpenJDK 64-Bit Server VM (build 23+-google-release-sts-681679659, mixed mode, sharing)
[   +1 ms] executing: /usr/bin/adb devices -l
[   +1 ms] ProcessManager.run('/usr/bin/adb devices -l')
[   +1 ms] Gallium device polling: connecting to galliumd.
[   +1 ms] [✓] Flutter (Channel google3, on Debian GNU/Linux rodete 6.9.10-1rodete5-amd64, locale en_US.UTF-8)
[   +1 ms]     • Framework revision 621e7ef951 (13 days ago), 2024-09-25T00:00:00.000
[   +1 ms]     • Engine revision d6d5fdba6a
[   +1 ms]     • Dart version dd73afd20b
[   +1 ms] [✓] Android toolchain - develop for Android devices (Android SDK version Stable)
[   +1 ms]     • Android SDK at google3
[   +1 ms]     • Platform Stable, build-tools Stable
[   +1 ms]     • Java binary at: /usr/local/buildtools/java/jdk/bin/java
[   +1 ms]     • Java version OpenJDK Runtime Environment (build 23+-google-release-sts-681679659)
[   +1 ms] [✓] VS Code (version 1.87.1)
[   +1 ms]     • VS Code at /usr/share/code
[   +1 ms]     • Flutter extension can be installed from:
[   +1 ms]       🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
[  +43 ms] List of devices attached
           localhost:38637        device product:oriole model:Pixel_6 device:oriole transport_id:5
[   +2 ms] Gallium device polling failed to connect to galliumd: gRPC Error (code: 14, codeName: UNAVAILABLE, message: Error connecting: SocketException: Connection refused (OS Error: Connection refused, errno = 111),
address = localhost, port = 46084, details: null, rawResponse: null, trailers: {})
[   +1 ms] /usr/bin/adb -s localhost:38637 shell getprop
[   +1 ms] ProcessManager.run('/usr/bin/adb -s localhost:38637 shell getprop')
[ +164 ms] ro.hardware = oriole
[   +2 ms] ro.build.characteristics = nosdcard
[   +1 ms] executing: uname -m
[   +1 ms] ProcessManager.runSync('uname -m')
[   +3 ms] Exit code 0 from: uname -m
[   +1 ms] x86_64
[   +1 ms] executing: /usr/bin/adb devices -l
[   +1 ms] ProcessManager.run('/usr/bin/adb devices -l')
[   +3 ms] List of devices attached
           localhost:38637        device product:oriole model:Pixel_6 device:oriole transport_id:5
[   +2 ms] [✓] Connected device (2 available)
[   +1 ms]     • Pixel 6 (mobile) • localhost:38637 • android-arm64 • Android 15 (API 35)
[   +1 ms]     • Linux (desktop)  • linux           • linux-x64     • Debian GNU/Linux rodete 6.9.10-1rodete5-amd64
[   +1 ms] [✓] Google3 (on linux)
[   +1 ms]     • KVM enabled
[   +1 ms] • No issues found!
[   +1 ms] "flutter doctor" took 306ms.
[ +646 ms] ensureAnalyticsSent: 645ms
[   +2 ms] Running 1 shutdown hook
[   +1 ms] Shutdown hooks complete
[   +1 ms] exiting with code 0
```

</details>

### Internal Bug ID
b/364625445

**Comment by TahaTesser** [comment]
@dsanagustin 
Thanks for the report!

**Comment by TahaTesser** [comment]
> Double tapping a tick should move the slider indicator to that tick.

I've tried both iOS and Android platforms for native Sliders and they don't move Slider thumb when double tapped. You'd need to swipe up or down to move the the thumb and adjust the Slider value.  This  also works in Flutter!

However, as you pointed there is indeed a bug here in Flutter Slider with double tap, it moves the thumb to zero Slider value which is unexpected.

**Comment by dsanagustin** [comment]
I am aware that swiping up or down is the way to move the slider in accessibility mode. But are you saying that the inability to double tap to a tick is working as intended?

**Comment by TahaTesser** [comment]
> I mention double tapping in accessibility mode because that is the equivalent of single tapping without accessibility mode. There, users can single tap to a specific tick on a slider, but accessibility users can't double tap to achieve that same behavior.

Single tap is on the `Slider` area in non-accessibility mode itself but accessibility mode doesn't require tapping on the widget area, you can double tap anywhere on the screen once you enable talkback or voice over. 

 In the accessibility, the Slider is only adjustable with swipe up or down gestures which is the same behavior on Android, double tap shouldn't the change slider value.

**Comment by dsanagustin** [comment]
Internal bug id: b/364625445

**Comment by TahaTesser** [comment]
Turns out a better fix is required from the engine side.  I've unassigned myself for someone with experience on the engine side to fix this issue. cc: @bleroux

**Comment by bleroux** [comment]
Reopening while waiting for [the Android fix](https://github.com/flutter/engine/pull/56452) to land.

**Comment by bleroux** [comment]
Closing as fixed, now that the [Android fix](https://github.com/flutter/engine/pull/56452) and the [iOS fix](https://github.com/flutter/engine/pull/56427) have landed.
